msbuild/dbuild/CompileD.cs

```diff
@@ -66,31 +106,508 @@ public virtual ITaskItem[] Sources
             }
         }
 
-        public string CommandLineTemplate
+        // Hidden
+        public bool DoNotLink
         {
-            get;
-            set;
+            get { return GetBoolProperty("DoNotLink"); }
+            set
+            {
+                SetBoolProperty("DoNotLink", "Do Not Link",
+                                "Compile only. Do not link (-c)",
+                                "-c", value);
+            }
         }
 
-        public bool ShowCommandLine
+        public string CodeGeneration
         {
-            get;
-            set;
+            get { return GetStringProperty("CodeGeneration"); }
+            set
+            {
+                string[][] switchMap = new string[3][]
+                {
+                    new string[2] { "32BitsMS-COFF", "-m32mscoff" },
+                    new string[2] { "32Bits", "-m32" },

```

Comment:
I avoided supporting OMF, so better not make it "first class", i.e. either remove the option or make it the specialized one "32Bits-OMF". This option is hidden anyway, though.

Reply:
Or is it needed for LDC? I suspect you use it to avoid dependencies between settings.

Reply:
Yes, -m32 is used by LDC. Do you see another way to implement this?
As you say, it is automatically set depending on the platform and the compiler.